src/Avalonia.Base/Data/BindingOperations.cs

```diff
@@ -33,9 +34,84 @@ public static class BindingOperations
 
             var mode = binding.Mode;
 
+            AvaloniaPropertyMetadata? metadata = null;
+
             if (mode == BindingMode.Default)
             {
-                mode = property.GetMetadata(target.GetType()).DefaultBindingMode;
+                metadata = property.GetMetadata(target.GetType());
+
+                mode = metadata.DefaultBindingMode;
+            }
+
+            var updateSourceTrigger = binding.UpdateSourceTrigger;
+
+            if (updateSourceTrigger == UpdateSourceTrigger.Default)
+            {
+                metadata ??= property.GetMetadata(target.GetType());
+
+                updateSourceTrigger = metadata.UpdateSourceTrigger;
+            }
+
+            static IObservable<object?> GetTargetPropertyObservable(
+                InstancedBinding binding,
+                IAvaloniaObject target,
+                AvaloniaProperty property,
+                UpdateSourceTrigger updateSourceTrigger)
+            {
+                var explicitSourceUpdate = binding
+                    .ExplicitUpdateRequested
+                    .Where(m => m == InstancedBinding.ExplicitUpdateMode.Source)
+                    .Select(_ => target.GetValue(property));
```

Comment:
I'm slightly worried about the performance overhead of adding this observable to every binding, but I've been unable to profile because the many of the benchmarks now throw.

Reply:
Yeah, we can probably write some simple (non-allocating) wrapper ourselves